File: java/algorithm/string/leetcode/ReverseString.java
package algorithm.string.leetcode;

/**
 * @author roseduan
 * 反转字符串
 */
public class ReverseString {

    public void reverseString(char[] s) {
        int i = 0, j = s.length - 1;
        while (i < j) {
            char temp = s[i]; s[i] = s[j]; s[j] = temp;
            i++; j--;
        }
    }
}

File: go/string/reverse_string.go
package string

// 反转字符串

func reverseString(s []byte) {
	i, j := 0, len(s)-1
	for i < j {
		s[i], s[j] = s[j], s[i]
		i, j = i+1, j-1
	}
}
